lightning-invoice/src/utils.rs

```diff
@@ -236,8 +239,10 @@ where L::Target: Logger {
 	fn find_route(
 		&self, payer: &PublicKey, params: &RouteParameters, _payment_hash: &PaymentHash,
 		first_hops: Option<&[&ChannelDetails]>, scorer: &S
-	) -> Result<Route, LightningError> {
-		find_route(payer, params, &*self.network_graph, first_hops, &*self.logger, scorer)
+		) -> Result<Route, LightningError> {
+		let mut locked_random_seed_bytes = self.random_seed_bytes.lock().unwrap();
+		*locked_random_seed_bytes = sha256::Hash::hash(&*locked_random_seed_bytes).into_inner();
+		find_route(payer, params, &*self.network_graph, first_hops, &*self.logger, scorer, &locked_random_seed_bytes.clone())
```

Comment:
As is the lock is held through `find_route` which we maybe don't want. Instead, we'll need an explicit `{}` scope to tell the compiler where to drop the `locked_random_seed_bytes`

Reply:
Should we consider parameterizing `DefaultRouter` by `K: Deref where K::Target: KeysInterface` and call `get_secure_random_bytes` each time? I suppose the interface is too broad, but I wonder if a narrower `Randomness` interface is warranted. No strong opinion here, but it would avoid the mutex.

Reply:
I now went the scoping route with https://github.com/lightningdevkit/rust-lightning/commit/386a0b553130173f576ecc89424fcd6791a41dcc. However, I agree that introducing a `Randomness` interface (not only here) would be a nicer abstraction than pushing around arrays with `random_seed_bytes`. Possibly a good follow-up PR?

Reply:
I suppose we could, yea, agree we'd want to do it in a followup, though. I'm a little hesitant to replace a trivial parameter with a full interface, though, as it just seems like overkill, but we can discuss that later.